tensorflow/contrib/tensorrt/test/reshape_transpose_test.py

```diff
@@ -0,0 +1,183 @@
+        identity = array_ops.identity(reshape, "identity")
+      array_ops.identity(identity, name=output_name)
+    return trt_test.TfTrtIntegrationTestParams(
+        gdef=g.as_graph_def(),
+        input_names=[input_name],
+        input_dims=[input_dims],
+        output_names=[output_name],
+        expected_output_dims=[(100, 24*24*2)])
+
+  def ExpectedEnginesToBuild(self, run_params):
+    """Return the expected engines to build."""
+    return ["my_trt_op_0"]
+
+class ReshapeInverseTest(trt_test.TfTrtIntegrationTestBase):
+
+  def GetParams(self):
+    """Create a graph containing single segment."""
+    dtype = dtypes.float32
+    input_name = "input"
+    input_dims = [100, 24, 24, 2]
+    output_name = "output"
+    g = ops.Graph()
+    with g.as_default():
+      inp = array_ops.placeholder(
+          dtype=dtype, shape=[None] + input_dims[1:], name=input_name)
+      with g.device("/GPU:0"):
+        reshape = array_ops.reshape(inp, [-1, 24*24*2])
+        reshape = array_ops.reshape(reshape, [-1, 24, 24, 2])
+        identity = array_ops.identity(reshape, "identity")
+      array_ops.identity(identity, name=output_name)
+    return trt_test.TfTrtIntegrationTestParams(
+        gdef=g.as_graph_def(),
+        input_names=[input_name],
+        input_dims=[input_dims],
+        output_names=[output_name],
+        expected_output_dims=[(100, 24, 24, 2)])
+
+  def ExpectedEnginesToBuild(self, run_params):
+    """Return the expected engines to build."""
+    return ["my_trt_op_0"]
+
+class ManyReshapeTest(trt_test.TfTrtIntegrationTestBase):
+
+  def GetParams(self):
+    """Create a graph containing single segment."""
+    dtype = dtypes.float32
+    input_name = "input"
+    input_dims = [100, 24, 24, 2]
+    output_name = "output"
+    g = ops.Graph()
+    with g.as_default():
+      inp = array_ops.placeholder(
+          dtype=dtype, shape=[None] + input_dims[1:], name=input_name)
+      with g.device("/GPU:0"):
+        reshape = array_ops.reshape(inp, [-1, 24*24, 2])
+        reshape = array_ops.reshape(reshape, [-1, 24*2, 24])
+        reshape = array_ops.reshape(reshape, [-1, 24, 24*2])
+        reshape = array_ops.reshape(reshape, [-1, 6, 4, 24, 2])
+        reshape = array_ops.reshape(reshape, [-1, 6, 4, 6, 4, 2])
+        reshape = array_ops.reshape(reshape, [-1, 6, 4, 6, 4, 2, 1])
+        reshape = array_ops.reshape(reshape, [-1, 24, 24, 2])
+        identity = array_ops.identity(reshape, "identity")
+      array_ops.identity(identity, name=output_name)
+    return trt_test.TfTrtIntegrationTestParams(
+        gdef=g.as_graph_def(),
+        input_names=[input_name],
+        input_dims=[input_dims],
+        output_names=[output_name],
+        expected_output_dims=[(100, 24, 24, 2)])
+
+  def ExpectedEnginesToBuild(self, run_params):
+    """Return the expected engines to build."""
+    return ["my_trt_op_0"]
+
+class SimpleTransposeTest(trt_test.TfTrtIntegrationTestBase):
```

Comment:
As mentioned above please add tests for the case where first element of the permutation is not 0.